packages/checkbox-group/src/vaadin-checkbox-group.js

```diff
@@ -352,6 +347,29 @@ class CheckboxGroup extends FieldMixin(FocusMixin(DisabledMixin(ElementMixin(The
     }
   }
 
+  /** @private */
+  __onCheckboxChange() {
+    this.dirty = true;
+
+    // Normally, the validation is triggered on `checked-changed` (in the value property observer).
+    // However, it can be disabled, in which case we need to trigger it manually here.
+    if (!this.__shouldValidateOnProgrammaticValueChange()) {
+      this.validate();
+    }
+  }
+
+  /**
+   * Override this method to define whether the component should validate on
+   * programmatic value property change.
+   *
+   * WARNING: Do not rely on this method because it will be removed later.
+   *
+   * @private
+   */
+  __shouldValidateOnProgrammaticValueChange() {
+    return true;
```

Comment:
Note (just fyi): Because I only see this implemented for CheckboxGroup, RadioGroup and Select. 

There was a recent discussion on discord (flow) that somebody doesn't used the Binder to build its form and to clear the form after successful creation he used e.g. `TextField::clear` this also resulted in invalid fields all over the place. 

Workaround I have provided was: Call `clear()` twice. "Problem gone" - just something to keep in mind, because it effects not just those components, but all.